metadata-ingestion/src/datahub/ingestion/source/tableau/tableau.py

```diff
@@ -835,13 +842,18 @@ def __init__(
         self.ctx: PipelineContext = ctx
         self.platform = platform
 
-        self.site: Optional[SiteItem] = site
-        if site_id is not None:
-            self.site_id: str = site_id
+        self.site: Optional[SiteItem] = None
+        if isinstance(site, SiteItem):
+            self.site = site
+            assert site.id is not None, "Site ID is required"
+            self.site_id = site.id
+            self.site_content_url = site.content_url
+        elif isinstance(site, SiteIdContentUrl):
+            self.site = None
+            self.site_id = site.site_id
+            self.site_content_url = site.site_content_url
```

Comment:
Looks like `site_content_url` is also defined as @property below which could be removed, to avoid confusion.

Reply:
indeed
done!